watcher/local/adjudicatorpubsub.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2021 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package local
+
+import (
+	"github.com/pkg/errors"
+
+	"perun.network/go-perun/channel"
+	"perun.network/go-perun/pkg/sync/atomic"
+	"perun.network/go-perun/watcher"
+)
+
+var _ watcher.AdjudicatorSub = &adjudicatorPubSub{}
+
+const adjPubSubBufferSize = 10
+
+type (
+	adjudicatorPubSub struct {
+		closed atomic.Bool
+		pipe   chan channel.AdjudicatorEvent
+	}
+
+	adjudicatorPub interface {
```

Comment:
Yes, I agree that interfaces should be introduced in the PRs using them.

But, in this case, I had created a PR including all the changes for the watcher and then split it up into 5 parts. After splitting it turned out that, earlier PR splits (1/5) has interfaces that will be used in later splits (3/5).

Reorganizing all the commits such that each PR is self contained would require a lot of effort. So, I think if you are okay with it, we ignore the linter warning for unused.

This shouldn't be a problem because, we are merging only into a feature branch, and all the tests, linter checks will pass on the feature branch when all 5 split PRs are merged.